keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/CheckScheduler.kt

```diff
@@ -84,6 +86,27 @@ class CheckScheduler(
     }
   }
 
+  @Scheduled(fixedDelayString = "\${keel.scheduled.agent.frequency:PT1M}")
+  fun invokeAgent() {
+    if (enabled) {
+      agentLockRepository.agents.forEach {
+        val agentName: String = it.javaClass.simpleName
+        val lockAcquired = agentLockRepository.tryAcquireLock(agentName, it.lockTimeoutSeconds)
+        if (lockAcquired) {
+          val job = launch {
+            it.invokeAgent()
+          }
+          runBlocking {
+            job.join()
```

Comment:
Do you want to wait for all launched jobs for all agents to end before completing?